kore/src/Kore/Step/Axiom/Matcher.hs

```diff
@@ -687,6 +696,39 @@ matchNormalizedAc pushValue wrapTermLike normalized1 normalized2
         _ -> empty
     lift $ Foldable.traverse_ pushValue concrete12
     lift $ Foldable.traverse_ pushValue abstractMerge
+  | otherwise = do
```

Comment:
I'm not sure this should be an `otherwise` guard. The case we want to cover here is: The first Map is a single element (the key and value are both variables) and one opaque map (a variable). The second Map may be anything.

If the single key in the first Map is equal to any key in the second Map, then the corresponding values must match, and `frame1` must match the remainder of the second Map.

If the single key is not in the second Map, then match it with any (concrete or symbolic) key in the second map, and match the corresponding values. `frame1` also matches the remainder of the second Map.